consensus/istanbul/validator/weighted.go

```diff
@@ -1,979 +0,0 @@
-	var (
-		newWeightedValidators []*weightedValidator
-		newWeightedDemoted    []*weightedValidator
-		govNode               *weightedValidator
-		newValidatorsStaking  []float64
-		newDemotedStaking     []float64
-		govNodeStaking        float64
-	)
-	for idx, val := range stakingAmounts {
-		if isSingleMode && govNodeAddr == weightedValidators[idx].Address() {
-			govNode = weightedValidators[idx]
-			govNodeStaking = val
-		} else if uint64(val) >= minStaking {
-			newWeightedValidators = append(newWeightedValidators, weightedValidators[idx])
-			newValidatorsStaking = append(newValidatorsStaking, val)
-		} else {
-			newWeightedDemoted = append(newWeightedDemoted, weightedValidators[idx])
-			newDemotedStaking = append(newDemotedStaking, val)
-		}
-	}
-
-	// when no validator has more than minimum staking amount of KAIA, all members are in validators
-	if len(newWeightedValidators) <= 0 {
-		// 1. if governance mode is not single,
-		// 2. if governance mode is single and governing node does not have minimum staking amount of KAIA as well
-		if !isSingleMode || uint64(govNodeStaking) < minStaking {
-			newWeightedValidators, newValidatorsStaking = newWeightedDemoted, newDemotedStaking
-			newWeightedDemoted, newDemotedStaking = []*weightedValidator{}, []float64{}
-			logger.Debug("there is no council member staking more than the minimum, so all become validators", "numValidators", len(newWeightedValidators), "isSingleMode", isSingleMode, "govNodeAddr", govNodeAddr, "govNodeStaking", govNodeStaking, "minStaking", minStaking)
-		}
-	}
-
-	// if the governance mode is single, governing node is added to validators all the time.
-	if isSingleMode {
-		newWeightedValidators = append(newWeightedValidators, govNode)
-		newValidatorsStaking = append(newValidatorsStaking, govNodeStaking)
-	}
-	return newWeightedValidators, newValidatorsStaking, newWeightedDemoted, newDemotedStaking
-}
-
-// getStakingAmountsOfValidators calculates stakingAmounts of validators.
-// If validators have multiple staking contracts, stakingAmounts will be a sum of stakingAmounts with the same rewardAddress.
-//   - []*weightedValidator : a list of validators which type is converted to weightedValidator
-//   - []float64 : a list of stakingAmounts.
-func getStakingAmountsOfValidators(validators istanbul.Validators, stakingInfo *reward.StakingInfo) ([]*weightedValidator, []float64, error) {
-	nVals := len(validators)
-	weightedVals := make([]*weightedValidator, nVals)
-	stakingAmounts := make([]float64, nVals)
-
-	for vIdx, val := range validators {
-		weightedVal, ok := val.(*weightedValidator)
-		if !ok {
-			return nil, nil, errors.New(fmt.Sprintf("not weightedValidator. val=%s", val.Address().String()))
-		}
-		weightedVals[vIdx] = weightedVal
-
-		if cIdx, err := stakingInfo.GetIndexByNodeAddress(weightedVal.address); err == nil {
```

Comment:
We can delete most methods of the old `reward.StakingInfo` type.

Reply:
Removed it. Since most of the functionality has been moved to kaiax and only the `reward.staking_info` struct remains for the downloader, it seems better to move it to either downloader or kaiax and remove the `reward` package.

Reply:
Yes, good idea. I'd prefer kaiax/staking/staking_info.go:StakingInfoP2p. Having it under downloader/ might cause import cycle from node/cn/handler.go?

Reply:
node/cn/handler.go already importing downloader so cycle doesn't occured. However, it could confusing if staking_info structures are located in multiple locations. I'll put it under kaiax/staking as you mentioned.

Reply:
deleted reward package https://github.com/kaiachain/kaia/pull/190/commits/2544c6d36b14204e687685db933fe890d54ad963